### PR TITLE
Implemented @page-url unmasking

### PR DESCRIPTION
**Pull request by slde-gorillaman**
Closes: #106 
Replaces: #163

### CODE REVIEW

**Comment by slde-gorillaman** [comment]
There seems to be a problem with PHPUnit on Travis :/

**Comment by aik099** [comment]
> There seems to be a problem with PHPUnit on Travis :/

That's not any test in particular or PHPUnit itself. It's a problem in ParaTest library I use to run tests in groups in parallel. I've reported it on https://github.com/brianium/paratest/issues/209.

**Comment by aik099** [comment]
Please rebase, because in #166 I've fixed broken unit tests.

**Comment by slde-gorillaman** [comment]
Done :)

**Comment by aik099** [comment]
I'm a bit busy at work currently. Will try to review your changes as soon as I can. Apologies for unexpected delays.

**Comment by aik099** [comment]
Not sure how you did it (or maybe GitHub showing it wrong), but PR consists of 6 commits, from which:

* one is made by you
* 5 results of unusual rebase where author is me, but committer is you

What I want to see (git history wise) before PR is merged is PR consisting of 1 commit made off current `develop` branch last commit.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 ### Added
-...
+-- Added support for url parameter unmasking (e.g. `{param}` text in the url of `@page-url` annotation) by [@slde-gorillaman].
```
Comment (aik099) [inline CHANGELOG.md]: There is extra `-` in front. Should be 1, but there is 2.

**Comment by aik099** [comment]
What kind of repository setup you have:

GitHub Advertised:

1. forked `qa-tools/qa-tools` (original) into `slde-gorillaman/qa-tools` (fork)
2. cloned `slde-gorillaman/qa-tools` (fork) locally
3. created `feature/page-url-unmasking` branch in local repository
4. pushed `feature/page-url-unmasking` branch to the `origin` remote (resolves to `slde-gorillaman/qa-tools` (fork))
5. sent PR from `feature/page-url-unmasking` branch of the `slde-gorillaman/qa-tools` (fork)

Composer Advertised:

1. forked `qa-tools/qa-tools` (original) into `slde-gorillaman/qa-tools` (fork)
2. cloned `qa-tools/qa-tools` (original) locally
3. added `slde-gorillaman` remote to the local repository, that points to `slde-gorillaman/qa-tools` (fork)
4. created `feature/page-url-unmasking` branch in local repository
5. pushed `feature/page-url-unmasking` branch to the `slde-gorillaman` remote (resolves to `slde-gorillaman/qa-tools` (fork))
6. sent PR from `feature/page-url-unmasking` branch of the `slde-gorillaman/qa-tools` (fork)

The difference between 2 is that with `Composer Advertised` you:

* will have up to date `develop` (or `master`) branch right from original repository when doing `git pull`
* can review code from different remotes at once
* when creating new branch (for PR) from `develop` or `master` that would be `develop`/`master` of original repo, which is always up to date

With `GitHub Advertised`:

* you need to bring `develop` or `master` branch of your fork up to date to create new branch from it
* doing push will automatically put data in your fork

**Comment by aik099** [comment]
To me:

* doing a rebase is as easy as `git rebase develop`, because `develop` comes from `qa-tools/qa-tools`
* doing a squash is as easy as `git rebase -i HEAD~3` (3 is number of commits in a branch)

**Comment by aik099** [comment]
Your changes were incorporated into #191 PR. Thank you, @slde-gorillaman .

**Pull request closed without merging**